instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/Config.java

```diff
@@ -16,15 +16,15 @@
 import javax.annotation.Nullable;
 
 /**
- * Represents the global agent configuration consisting of system properties, environment variables,
- * contents of the agent configuration file and properties defined by the {@code
- * ConfigPropertySource} SPI implementations.
+ * Represents the global instrumentation configuration consisting of system properties and
+ * environment variables; and, if using the OpenTelemetry javaagent, contents of the agent
+ * configuration file and properties defined by the {@code ContextCustomizer} SPI implementations.
  *
  * <p>In case any {@code get*()} method variant gets called for the same property more than once
  * (e.g. each time an advice class executes) it is suggested to cache the result instead of
- * repeatedly calling {@link Config}. Agent configuration does not change during the runtime so
- * retrieving the property once and storing its result in a static final field allows JIT to do its
- * magic and remove some code branches.
+ * repeatedly calling {@link Config}. The instrumentation configuration does not change during the
+ * runtime so retrieving the property once and storing its result in a static final field allows JIT
+ * to do its magic and remove some code branches.
  */
 @AutoValue
 public abstract class Config {
```

Comment:
Is it possible to make this class just delegate to `ConfigProperties`? I realized we have some confusing SPI because we can customize `ConfigProperties` in SDK and `Config` here. If we need to expose a factory to construct `DefaultConfigProperties` for the case where there's no javaagent or SDK, we could probably figure something out, but I guess we do need to converte these SPIs somehow.

Reply:
Also I guess this is the last chance to look into possibly moving `ConfigProperties` to opentelemetry-api (existing one could be deprecated and just delegate to it). If we need config from instrumentation, having two, one for SDK and one for instrumentation, seems like it is making us struggle a lot.

Reply:
> possibly moving `ConfigProperties` to opentelemetry-api

That would be perfect, I think.

Reply:
> possibly moving ConfigProperties to opentelemetry-api

The idea being that instrumentation would only need to rely on `opentelemetry-api` instead of the additional `autoconfigure-spi` dependency?

Reply:
I'm not a fan of putting this interface into the OTel API. This is 100% an SDK construct, used only by the SDK. It shouldn't be added to the API.

Reply:
We currently have basically the exact same concept used in instrumentation as well as `Config` here, so currently have that duplication which is particularly confusing when customizing the config. Library instrumentation could be purely programmatic - I think we decided to fallback to system a while back but could reconsider this. But if we think that OTel instrumentation "is supposed to" support config through environment, then it stops being an SDK-only construct and we may want to think through more options on it.

@mateuszrzeszutek In the meantime, as with many of our classes, maybe we should just move `Config` to `internal` for now 😅

Reply:
Notes from today's special topics meeting:

1. We'll add hyphen normalization to SDK's `ConfigProperties`
2. `ConfigProperties` will be extracted to its own, separate `opentelemetry-config` module in the SDK repo. It'll be used both by agent and the SDK autoconfigure. The global `Config` instance will be moved from `instrumentation-api` to the `javaagent-extension-api` bootstrap part (it still needs to be available in the agent), `Instrumenter`s and telemetry builders will have a new method that allows passing a `ConfigProperties` instance.
3. In case 1 & 2 are not finished before the release, we'll just move `Config` to internal so that it does not block marking the `instrumentation-api` release as RC.